lib/pure/uri.nim

```diff
@@ -311,6 +312,36 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc parseQuery*(query: string): TableRef[string, seq[string]] = 
+  ## parses the query part of an uri.
+  if query == "":
+    return
+  result = newTable[string, seq[string]]()
+  for part in query.split("&"):
```

Comment:
The cgi module shows how to the parsing efficiently. ``split`` is often a code smell.